File: Data Structures & Algorithms/palindromic-substrings/submission-2.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        if len(s) == 1:
            return 1;

        ret = 0;
        i = 0
        while i < len(s):
            ret += 1;
            left = i - 1;
            right = i + 1;

            while (left > -1 and right < len(s)):
                if (s[left] == s[right]):
                    left -= 1;
                    right += 1;
                    ret += 1;
                else: 
                    break;
            
            left = i;
            right = i + 1;
            while (left > -1 and right < len(s)):
                if (s[left] == s[right]):
                    left -= 1;
                    right += 1;
                    ret += 1;
                else: 
                    break;

            i += 1;

        return ret;


        

File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> ret = vector<int>(temperatures.size());
        for (int i = 0; i < temperatures.size(); i++) {
            for (int j = i + 1 ; j < temperatures.size(); j++) {
                if (temperatures[j] > temperatures[i]) {
                    ret[i] = j - i;
                    break;
                }
            }
        }
        return ret;
    }
};
